script.js

```diff
@@ -1,4 +1,289 @@
+    console.log(dealerHand);
+
+    // change the gameMode
+    currentGameMode = GAME_CARDS_DRAWN;
+
+    // write and return appropriate output message
+    outputMessage = `Everyone has been dealt a card. Click submit to evaluate cards`;
+    return outputMessage;
+  }
+  // SECOND CLICK OF SUBMIT BUTTON
+  if (currentGameMode == GAME_CARDS_DRAWN) {
+    // check for blackjack
+    var playerHasBlackjack = checkForBlackjack(playerHand);
+    var dealerHasBlackjack = checkForBlackjack(dealerHand);
+    console.log(`Does player have blackjack? ${playerHasBlackjack}`);
+    console.log(`Does dealer have blackjack? ${dealerHasBlackjack}`);
+    if (playerHasBlackjack == true || dealerHasBlackjack == true) {
+      // both player and dealer has blackjack -> tie
+      if (playerHasBlackjack == true && dealerHasBlackjack == true) {
+        outputMessage =
+          displayPlayerAndDealerHands(playerHand, dealerHand) +
+          `<br>Blackjack tie!`;
+      }
+      // only player has blackjack -> player wins
+      else if (playerHasBlackjack == true && dealerHasBlackjack == false) {
+        outputMessage =
+          displayPlayerAndDealerHands(playerHand, dealerHand) +
+          `<br>Player wins by blackjack!`;
+      }
+      // only dealer has blackjack -> dealer wins
+      else {
+        outputMessage =
+          displayPlayerAndDealerHands(playerHand, dealerHand) +
+          `<br>Dealer wins by blackjack!`;
+      }
+      console.log(outputMessage);
+    } else {
+      // no blackjack -> game continues
+
+      // calculate total hand value of both player and dealer
+      var playerHandTotalValue = calculateTotalHandValue(playerHand);
+      var dealerHandTotalValue = calculateTotalHandValue(dealerHand);
+      console.log(`Player total hand value = ${playerHandTotalValue}`);
+      console.log(`Dealer total hand value = ${dealerHandTotalValue}`);
+      // compare total hand value
+      // same value -> tie
+      if (playerHandTotalValue == dealerHandTotalValue) {
+        outputMessage =
+          displayPlayerAndDealerHands(playerHand, dealerHand) +
+          `It is a tie!<br>` +
+          displayHandTotalValues(playerHandTotalValue, dealerHandTotalValue);
+      }
+      // player higher value -> player wins
+      else if (playerHandTotalValue > dealerHandTotalValue) {
+        outputMessage =
+          displayPlayerAndDealerHands(playerHand, dealerHand) +
+          `<br>Player wins!<br>` +
+          displayHandTotalValues(playerHandTotalValue, dealerHandTotalValue);
+      }
+      // dealer higher value -> dealer wins
+      else {
+        outputMessage =
+          displayPlayerAndDealerHands(playerHand, dealerHand) +
+          `<br>Dealer wins!<br>` +
+          displayHandTotalValues(playerHandTotalValue, dealerHandTotalValue);
+      }
+      // change the gameMode
+      currentGameMode = GAME_RESULTS_SHOWN;
+
+      // appropriate output message
+      return outputMessage;
+    }
+  }
```

Comment:
Personal preference of mine is to abstract away most of the main into multiple helper functions so tha it's easier for both yourself and a code reviewer to catch what's going on at a glance. The code might end up looking something like: |

`
var main = function (input) {
if (currentGameMode == GAME_START) {
  gameStart()
}

else if (currentGameMode == GAME_CARDS_DRAWN) {
  showCardsDrawn()
}

else {
displayWinner()
}
`